src/supabase/queries/cases.ts

```diff
@@ -0,0 +1,85 @@
+import { Case, CaseUid, UserUid } from '../../types/types';
+import supabase from '../createClient';
+
+/**
+ * Fetch an array of `CaseId`s associated with a specific user. Fetches ids from `status` table.
+ *
+ * @param userId user Uuid
+ * @returns array of `CaseId`s
+ */
+export async function getCaseIdsFromUserId(
+  userId: UserUid,
+): Promise<CaseUid[]> {
+  try {
+    // fetch caseIds that match the specified userId
+    const { data } = await supabase
+      .from('status')
+      .select('caseId')
+      .eq('userId', userId);
+
+    // throw error if supabase data is empty
+    if (!data) {
+      throw new Error(`no caseIds found for the given user: ${userId}`);
+    }
+
+    // cast raw sql data as an array of CaseIds
+    return data.map(item => item.caseId as CaseUid);
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.warn('(getCaseIdsFromUserId)', error);
+    throw error;
+  }
+}
+
+/**
+ * Fetch an array of Case objects contained in an array of `CaseId`s. Fetches cases from `cases` table.
+ *
+ * @param caseIds list of `CaseId`s
+ * @returns array of `CaseCard` objects
+ */
+export async function getCasesByIds(caseIds: CaseUid[]): Promise<Case[]> {
+  try {
+    // query cases contained in a list of caseIds
+    const { data } = await supabase.from('Cases').select().in('id', caseIds);
+
+    // throw error if supabase data is empty
+    if (!data) {
+      throw new Error('no cases found');
+    }
+
+    // cast raw sql data as CaseCardProps data type
+    return data.map(item => {
+      const formattedCase: Case = {
```

Comment:
Sorry one little nit I missed but let's make this its own function (like parseCase or something around those lines) in case we need to use this in other queries later on!